.coderabbit.yaml

```
@@ -0,0 +1,44 @@
+      - **Assertions**: Assertions must be specific and provide informative messages on failure.
+  - path: "**/*.go"
+    instructions: |
+      Review the code as if you are a friendly but pedantic very senior developer in our team. We generally expect the following:
+      - **Clarity and Simplicity**: Write idiomatic Go. Code should be self-documenting. Complexity should be justified.
+      - **Concurrency**: Ensure goroutine safety. Use channels for communication and synchronization. Avoid race conditions.
+      - **Error Handling**: No silent failures. Errors must be checked, wrapped with context, and propagated. Use `errors.Is` and `errors.As`.
+      - **Resource Management**: Prevent resource leaks. Defer `close()` or `cleanup()` calls immediately after allocation. Manage goroutine lifecycles.
+      - **API Design**: APIs (REST, gRPC) must be well-defined, versioned, and backward-compatible. Input validation is mandatory.
+      - **Observability**: Instrument code with structured logging, metrics (e.g., Prometheus), and tracing. Logs should provide context, not just messages.
+      - **Security**: Be vigilant against vulnerabilities (e.g., injection, insecure direct object references). Sanitize inputs. Do not hardcode secrets.
+      - **Performance**: Write efficient code, but avoid premature optimization. Be mindful of algorithmic complexity and memory allocations.
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)